ambry-router/src/main/java/com/github/ambry/router/SimpleOperationTracker.java

```diff
@@ -248,6 +248,18 @@ class SimpleOperationTracker implements OperationTracker {
         // Here set the success target to the number of eligible replicas.
         replicaSuccessTarget = eligibleReplicas.size();
         break;
+      case ReplicateBlobOperation:
+        // Replicate one blob. Use the same "success target" and "request parallelism" config settings as PutOperation.
+        // Only difference is for ReplicateBlobOperation, crossColoEnabled is true.
+        // ON_DEMAND_REPLICATION_TODO: we may consider add its own configure setting if needed.
+        eligibleReplicas = getEligibleReplicas(datacenterName, EnumSet.of(ReplicaState.STANDBY, ReplicaState.LEADER));
```

Comment:
if you want to enable cross-colo, then getEligibleReplica's first argument has to be null. You can't specify a datacenter name as the first argument since it will only get the replicas from that specific datacenter.

Reply:
Good catch. Thank you.
Due to NonBlockRouterTest set routerConfig.routerGetEligibleReplicasByStateEnabled to false, so all my test works as expected. It uses all the 9 replicas and parallelism is 2.

Will fix it and add test coverage of NonBlockingRouterTest.